### ISSUE TITLE
Inline diagnostics in the editor do not get cleared up

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#65503]
**This is regression in 17.5 dogfood builds, does not repro on 17.4 release builds**

**Version Used**: 

```
Microsoft Visual Studio Enterprise 2022
Version 17.5.0 Preview 2.0 [33116.440.main]
VisualStudio.17.IntPreview/17.5.0-pre.2.0+33116.440.main
Microsoft .NET Framework
Version 4.8.09032
```

**Steps to Reproduce**:

1. Check the below Tools Option checkbox:
![image](https://user-images.githubusercontent.com/10605811/202673479-d6f63b25-e0b2-4c69-9b69-81e54caa81c9.png)

2. Open Roslyn.sln
3. Open source file Compilation.cs in Microsoft.CodeAnalysis.csproj
4. Goto line 115 and remove the semicolon at the end of the statement to introduce a compiler error
5. Wait for a few seconds to see the error list, squiggle and the inline diagnostic to appear
6. Undo the above step to add back the semicolon

**Expected Behavior**:
Error list, squiggle and inline diagnostic all refresh

**Actual Behavior**:
Error list refreshes fast, squiggle refreshes after a big delay (see #65502), but the inline diagnostic never refreshes and hangs around

![StaleInlineDiagnostic](https://user-images.githubusercontent.com/10605811/202674439-e60a327c-b3a2-4b9d-9d38-4156eeb08fda.gif)

**Comment by jasonmalinowski** [comment]
I've been running into this too.

**Comment by CoolCoderSuper** [comment]
Happens to me as well

**Comment by mavasani** [comment]
This is being reported by many customer feedbacks on 17.4, so we need to evaluate any fix here for 17.4 servicing.

**Comment by mavasani** [comment]
Reported in https://developercommunity.visualstudio.com/t/Bug---Inline-Error-Reporting-tooltip-s/10172624

**Comment by sujayvsarma** [comment]
> Reported in https://developercommunity.visualstudio.com/t/Bug---Inline-Error-Reporting-tooltip-s/10172624

As the reporter of that particular bug -- if you see the issue history on Dev Community, it was first reported by me for build `17.3.6 PREVIEW`. I can CONFIRM that this issue still exists in the current Preview as well as in the `RTM 17.4.1`.

Also, as I noted in my last comment on the dev community ticket, the issue is so horrible on the Preview build that I've had to stop using it completely. 

As a side note -- **all** LSP based analyzers are showing this behavior for me. That is, if I turn on LSP based experiences for HTML or CSS, the same happens there also.

**Comment by mavasani** [comment]
Reported in https://developercommunity.visualstudio.com/t/Editor-reports-errors-when-there-are-non/10213836

**Comment by mavasani** [comment]
Reported in https://developercommunity.visualstudio.com/t/Code-editor-errors-wont-go-away-when-fi/10224081

**Comment by mavasani** [comment]
@akhera99 Did we port this fix to 17.4 servicing?

**Comment by jasonmalinowski** [comment]
@mavasani Looks like we're still waiting on this: https://github.com/dotnet/roslyn/pull/65727
